### ISSUE TITLE
Vulkan Support added to Doom 3 BFG

### ISSUE THREAD

**Issue by AerowynX** [KozGit/DOOM-3-BFG-VR#198]
IDsoftware programmer added in Vulcan support recently https://github.com/DustinHLand/vkDOOM3/releases wonder if this would work well with this mod

**Comment by KozGit** [comment]
This looks pretty cool.  What I'm hoping happens is that once all the Vulkan issues are sorted, this is integrated into Robert Beckebans DOOM 3 BFG Port. ( https://github.com/RobertBeckebans/RBDOOM-3-BFG This is the port of D3BFG both VR mods used as a base. )  I really need to get this back in sync with that version, so I could kill two birds with one stone.

**Comment by RobertBeckebans** [comment]
I asked the vkDOOM3 developer on Twitter to upload the source so I can merge it and here we go :)
However vkDOOM3 has bigger renderer architecture changes. I will merge it within the next weekends but first I need to move a lot of my old stuff in the renderer. We suddenly have things like idRendererBackend instead of just some RB_ functions.
I need to merge all that stuff manually but I will do it so I finally have an excuse to get into Vulkan programming. It is a very nice learning excercise.

**Comment by KozGit** [comment]
Straight from the man himself - awesome!  Do you think the plan is to have separate OpenGL and Vulkan branches, or will both paths be implemented and user selected?

**Comment by AerowynX** [comment]
Awesome

On Aug 15, 2017 11:12 AM, "KozGit" <notifications@github.com> wrote:

Straight from the man himself - awesome! Do you think the plan is to have
separate OpenGL and Vulkan branches, or will both paths be implemented and
user selected?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/KozGit/DOOM-3-BFG-VR/issues/198#issuecomment-322497138>,
or mute the thread

**Comment by RobertBeckebans** [comment]
Yes. vkDOOM3 allows GL and VK as different target builds so you can compare both renderers.
It is necessary anyway because RBDOOM became the default D3 BFG Linux version and is shipped by several Distributions. Vulkan Linux drivers are way behind the propietary ones.
My plan is to add Vulkan so the Linux site Phoronix can add it to its GL vs. VK benchmark suite.

**Comment by RobertBeckebans** [comment]
BTW I shared a video about this fork on Twitter yesterday. It is really impressive what it has become even though I'm not into VR myself because I'm way to sensitive to motion sickness.

This is the video I found: https://www.youtube.com/watch?v=83kdr7D-DJ0

**Comment by AerowynX** [comment]
@Robert have you tried a Vive in full roomscale and only played teleport style movement? i have demoed to many who are super sensitive to motion sickness and had no problems on the Vive as long as it was teleport based movement
